### PR TITLE
Change 'request' event returned type to Promise<any[]>

### PR DESCRIPTION
**Pull request by ghost**
The returned type is at the moment a fixed array. Changing it to a promise will allow more flexibility.

Cheers.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Typescript definition file
 declare module 'fetch-intercept' {
   export interface FetchInterceptor {
-    request?(url: string, config: any): any[];
+    request?(url: string, config: any): Promise<any[]>;
```
Comment (mlegenhausen) [inline src/index.d.ts]: I think we need to allow both

```ts
request?(url: string, config: any): Promise<any[]> | any[];
```

**Review by mlegenhausen — CHANGES_REQUESTED** [review]

File: src/index.d.ts (hunk 1 above)
Comment (sjmcdowall) [reply src/index.d.ts]: I think that is correct .. both sounds more proper

Comment (aluczak) [reply src/index.d.ts]: Ok, just added the additional any[]

**Comment by mlegenhausen** [comment]
@aluczak Thanks.

**Comment by Harmonickey** [comment]
Please review my new issue regarding this change, I am getting an error trying to use the new Promise version of 'request' #26

**Pull request merged**